### DISCUSSION TITLE
"Relay Access Denied" Error When Sending Emails in Apple Mail

### DISCUSSION THREAD

**Discussion by minacle** [docker-mailserver/docker-mailserver#4219] [Q&A]
After following the [Usage](https://docker-mailserver.github.io/docker-mailserver/v14.0/usage/) page, including SPF, DKIM, and DMARC settings, I tried sending and receiving simple emails using Apple Mail. Receiving emails worked fine, but sending always resulted in a "Relay access denied" error. I adjusted various settings, such as the SMTP port, SSL/TLS options, and authentication method to password, but I still couldn't connect to the SMTP server.

Finding this issue peculiar, I installed Thunderbird to test sending and receiving emails, and everything worked perfectly without any issues.

I couldn't resolve this problem on my own. What could be the issue?

**Comment by casperklein** [comment]
Make sure to use one of those ports and not port 25:

https://github.com/docker-mailserver/docker-mailserver/blob/26a44995a9bba825b1348dbf3551687c2100f7a3/compose.yaml#L12-L15

**Comment by polarathene** [comment]
> I couldn't resolve this problem on my own. What could be the issue?

While unlikely, if you're macOS is quite old (4+ years without an update) it's possible that Apple Mail lacks support for compatible TLS cipher suites, but I think that would have been a different error. I know it was a problem in the past.

Relay access is denied if you are sending mail to DMS through port 25 which is only meant for delivery. Ports 465 (TLS) and 587 (STARTTLS) are intended for authenticated submission and is what your mail client should be using. These two ports require you to login with the DMS account credentials (`setup email add user@example.com secret-password`).

I don't use Apple Mail so I don't know how the settings differ from Thunderbird. Make sure you're connecting to the DMS FQDN/hostname (`mail.example.com`) which is often separate but possibly similar to your mail account domain (`user@example.com`). You should have some options about password/encryption types (authentication method?) like Thunderbird offers, one of those is plain-text and a separate setting for the port and protocol (_or they may be bundled into the same setting_), make sure 465 is TLS or 587 is STARTTLS.

Once you have that configured correctly, it should be fine. Sounds like you attempted to but are lacking information to why it failed to connect, did nothing appear in the DMS logs at all?

**Comment by minacle** [comment]
Thank you all for your responses.

I reset the docker-mailserver from scratch, then for some reason, I restarted the client Mac. After that, I added the account using the TLS SMTP port in Apple Mail, and it worked properly.

I'm not exactly sure which part made the difference, but here are my guesses:

1. **Reconfiguration of the Server**
   Initially, I configured my server as `example.com.` instead of `mail.example.com.`. This might have caused issues with Apple Mail.
2. **Restarting the Mac**
   Before restarting, changing the SMTP port and authentication method took a long time and often resulted in failure or hanging. Restarting might have resolved these issues, allowing the changes to apply correctly.
3. **DNS Propagation**
   Perhaps the DNS zone hadn't fully propagated since I configured it not long before. It might have simply been a matter of waiting and trying again later.

Whatever the issue was, it's working well now, and I have my own mail server up and running.
